cpp/src/slice2cpp/CPlusPlusUtil.cpp

```diff
@@ -20,95 +20,39 @@ using namespace IceUtilInternal;
 namespace
 {
 
-string toTemplateArg(const string& arg)
-{
-    if(arg.empty())
-    {
-        return arg;
-    }
-    string fixed = arg;
-    if(arg[0] == ':')
-    {
-        fixed = " " + fixed;
-    }
-    if(fixed[fixed.length() - 1] == '>')
-    {
-        fixed = fixed + " ";
-    }
-    return fixed;
-}
-
 string
-toOptional(const TypePtr& type, const string& scope, const StringList& metaData, int typeCtx)
+stringTypeToString(const TypePtr&, const StringList& metaData, TypeContext typeCtx)
 {
-    if (isProxyType(type))
+    string strType = findMetaData(metaData, typeCtx);
+
+    if (strType == "")
     {
-        // We map optional proxies like regular proxies, as optional<XxxPrx>.
-        return typeToString(type, scope, metaData, typeCtx);
+        strType = (typeCtx & TypeContext::UseWstring) != TypeContext::None ? "::std::wstring" : "::std::string";
     }
     else
     {
-        return "::std::optional<" + typeToString(type, scope, metaData, typeCtx) + '>';
+        assert(strType == "string" || strType == "wstring");
```

Comment:
Are you entrain this won't crash if the user does something like `cpp:type::foo`? I thought the validation of `cpp:type` was pretty minimal

Reply:
We do need to improve / fix the validation.